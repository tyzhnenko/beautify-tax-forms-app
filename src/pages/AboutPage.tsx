import { Typography } from "@material-tailwind/react";
import useTitle from "../utils/useTitle";

function AboutPage() {
  useTitle("Про проект");

  return (
    <div className="container w-auto px-10">
      <Typography as="article" className="pt-16">
        <h1 className="text-4xl text-center pb-10">Про проект</h1>

        <p className="pt-5">
          Якщо у вас є побажання чи ви знайшли помилки додайте їх опис{" "}
          <a
            className="underline"
            href="https://github.com/tyzhnenko/beautify-tax-forms-app/issues"
          >
            сюди
          </a>
        </p>
        <p className="pt-5">
          Пошта для звʼязку{" "}
          <a className="underline" href="mailto:beautifytaxforms@j4f.app">
            beautifytaxforms@j4f.app
          </a>
        </p>
      </Typography>
    </div>
  );
}

export default AboutPage;
