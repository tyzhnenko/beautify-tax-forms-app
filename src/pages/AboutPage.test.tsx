import { render, screen } from "@testing-library/react";
import AboutPage from "./AboutPage";
import "@testing-library/jest-dom";

describe("AboutPage component", () => {
  it("hould render the component", () => {
    render(<AboutPage />);

    expect(document.title).toEqual("Про проект");
    expect(
      screen.getByRole("heading", { name: "Про проект" })
    ).toBeInTheDocument();
    expect(
      screen.getByText(
        /Якщо у вас є побажання чи ви знайшли помилки додайте їх опис/i
      )
    ).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /сюди/i })).toHaveAttribute(
      "href",
      "https://github.com/tyzhnenko/beautify-tax-forms-app/issues"
    );
    expect(screen.getByText(/Пошта для звʼязку/i)).toBeInTheDocument();
    expect(
      screen.getByRole("link", { name: /beautifytaxforms@j4f.app/i })
    ).toHaveAttribute("href", "mailto:beautifytaxforms@j4f.app");
  });

  it("should match snapshot", () => {
    const { container } = render(<AboutPage />);
    expect(container).toMatchSnapshot();
  });
});
