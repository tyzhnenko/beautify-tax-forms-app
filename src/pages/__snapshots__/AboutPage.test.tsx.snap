// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage component should match snapshot 1`] = `
<div>
  <div
    class="container w-auto px-10"
  >
    <article
      class="block antialiased font-sans text-base font-light leading-relaxed text-inherit pt-16"
    >
      <h1
        class="text-4xl text-center pb-10"
      >
        Про проект
      </h1>
      <p
        class="pt-5"
      >
        Якщо у вас є побажання чи ви знайшли помилки додайте їх опис
         
        <a
          class="underline"
          href="https://github.com/tyzhnenko/beautify-tax-forms-app/issues"
        >
          сюди
        </a>
      </p>
      <p
        class="pt-5"
      >
        Пошта для звʼязку
         
        <a
          class="underline"
          href="mailto:beautifytaxforms@j4f.app"
        >
          beautifytaxforms@j4f.app
        </a>
      </p>
    </article>
  </div>
</div>
`;
